internal/httpcli/external.go

```diff
@@ -0,0 +1,52 @@
+package httpcli
+
+import (
+	"crypto/tls"
+	"net/http"
+	"reflect"
+	"sync"
+
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/schema"
+)
+
+type externalTransport struct {
+	base *http.Transport
+
+	mu        sync.RWMutex
+	config    *schema.TlsExternal
+	effective *http.Transport
+}
+
+func (t *externalTransport) RoundTrip(r *http.Request) (*http.Response, error) {
+	t.mu.RLock()
+	config, effective := t.config, t.effective
+	t.mu.RUnlock()
+
+	if current := conf.Get().ExperimentalFeatures.TlsExternal; current == nil {
+		return t.base.RoundTrip(r)
+	} else if reflect.DeepEqual(config, current) {
```

Comment:
```suggestion
	} else if !reflect.DeepEqual(config, current) {
		effective = t.update()
	}
```

and then you only need one `effective.RoundTrip(r)`?

I also think that it makes it clearer what happens here

Reply:
I like it. Will update in a follow-up PR. Branch cut is gonna happen and CI is slow atm so won't make it if I do it now :)